src/components/feed.jsx

```diff
@@ -19,11 +29,55 @@ const HiddenEntriesToggle = (props) => {
     </div>
   );
 };
+// Needed lifecycle events so replacing the functional component with class
+export default class Feed extends Component {
+  constructor(props) {
+    super(props);
+  }
+  current = -1;
+  previous = null;
+  handleKey = (e) => {
+    const nodeName = e.target.nodeName;
+    let node = this.postsDom.childNodes[this.current]; // current node
+    const l = this.postsDom.childNodes.length; // posts length
+    const keyCode = (window.event) ? e.which : e.keyCode;
+    if (nodeName !== 'INPUT' && nodeName !== 'TEXTAREA') {
+      if (keyCode === 74 || keyCode === 75) { // 74 -> j and 75 -> k
+        if (keyCode === 74 && this.current < l - 1) {
 
-export default (props) => {
-  const getEntryComponent = section => post => {
-    const isRecentlyHidden = (post.isHidden && (section === 'visible'));
+          this.current++;
+
+        } else if (keyCode === 75 && this.current > 0) {
+
+          this.current--;
 
+        }
+        if (this.previous) {
+
+          this.previous.style.borderColor = 'rgba(0,0,0,0)';
+
+        }
+        node = this.postsDom.childNodes[this.current];
+        if (node) {
+
+          node.style.borderColor = 'rgba(0,0,0,1)';
+          const y = window.pageYOffset + node.getBoundingClientRect().top;
+          scrollTo(element, y, 400);
```

Comment:
How about this?
```javascript
import scrollTo from 'animated-scrollto';
// it would be nice if `isEdge` and `rootElem` is evaluated just once and not every time we need to scroll something
const isEdge = window.navigator.userAgent.indexOf("Edge") > -1; 
let rootElem;

if (isEdge) {
  rootElem = document.body;
} else {
  rootElem = document.documentElement;
}
function animatedScroll(elem, duration, styles) { // should we add styles here?
   if(styles) {
      elem.style = {...elem.style, ...styles};
   }
   const y = windowlpageYOffset + elem.getBoundingClientRect().top;
   scrollTo(rootElem, y, duration)
}

```
If the other places that use window.scrollTo or window.scrollBy need animation, then it would be great to implement this there too.

Reply:
yup. makes sense.
put it in some `utils` file and import `animatedScroll` from it. so it doesn't clutter component files